tests/component-tests/pager/js/PagedTableTests.js

```diff
@@ -150,60 +159,113 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
     };
 
     fluid.tests.tooltipModuleSource = function (pager) {
-        var pageLinksTop = $("a", pager.pagerBar.locate("pageLinks"));
-        var pageLinksBottom = $("a", pager["pagerBar-1"].locate("pageLinks"));
-
-        var tooltipContents = [
-            [
-                {nodeName: "b", nodeText: "bird"},
-                {nodeName: "b", nodeText: "fish"}
-            ]
-        ];
-
         var sequence = [];
 
-        var assertVisibleTips = function (message, targetIds) {
-            sequence.push({
-                event: "{trackTooltips}.events.notifyFocusChange",
-                listener: function () {
-                    fluid.tests.tooltip.assertVisible(message, pager, targetIds, null, function (tooltip) {
-                        jqUnit.assertNode("The contents of the tooltip should be set", tooltipContents[0], $("b", tooltip));
-                    });
+        /* Change the page size and verify whether the tooltips are synced accordingly */
+
+        fluid.tests.setPageSize(pager, sequence, 2);
+        fluid.tests.verifyToolTips(pager, sequence, 2, [
+            [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "cat"} ],
+            [ {nodeName: "b", nodeText: "bird"}, {nodeName: "b", nodeText: "fish"} ],
+            [ {nodeName: "b", nodeText: "camel"}, {nodeName: "b", nodeText: "dragon"} ],
+            [ {nodeName: "b", nodeText: "ant"}, {nodeName: "b", nodeText: "ant"} ]
+        ]);
+
+        fluid.tests.setPageSize(pager, sequence, 3);
+        fluid.tests.verifyToolTips(pager, sequence, 3, [
+            [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "bird"} ],
+            [ {nodeName: "b", nodeText: "fish"}, {nodeName: "b", nodeText: "dragon"} ],
+            [ {nodeName: "b", nodeText: "ant"}, {nodeName: "b", nodeText: "ant"} ]
+        ]);
+
+        fluid.tests.setPageSize(pager, sequence, 4);
+        fluid.tests.verifyToolTips(pager, sequence, 4, [
+            [ {nodeName: "b", nodeText: "dog"}, {nodeName: "b", nodeText: "fish"} ],
+            [ {nodeName: "b", nodeText: "camel"}, {nodeName: "b", nodeText: "ant"} ]
+        ]);
+        
+        return {
+            name: "Pager tooltip tests",
+            tests: [
+                {
+                    name: "Tooltip visibility and contents",
+                    sequence: sequence
                 }
-            });
+            ]
         };
+    };
 
-        var tooltipTest = function (location) {
-            return function (idx, linkEl) {
-                var linkId = linkEl.id;
-                var link = $(linkEl);
-                sequence.push({
-                    element: link,
-                    jQueryTrigger: "focus"
+    fluid.tests.assertVisibleTips = function (pager, sequence, message, targetIds, expectedTooltip) {
+        sequence.push({
```

Comment:
In the old implementation the modification of the sequence was all done within the scope of a larger function. However, now that you have factored out this function and added new ones they are available publicly. This is good, but we need to make sure that the external api is expected. As I mentioned before, these all modify the sequence array directly and could cause an issue for someone who isn't expecting that. I wouldn't for example expect something called assertVisibleTips to modify a sequence array. I would expect it to perform some test to make sure that the expected tooltips are visible. I'd suggest a couple things, both here and for your other functions:

- Rename the functions to indicate what they are actually doing.
- Consider not modifying the sequence in place, but creating a new one and returning it.